submissions/ivan-chukhalo/frogger-game/app.js

```diff
@@ -0,0 +1,116 @@
+const AVAILABLE_ROWS_FOR_ENEMIES = [63, 146, 229]; 
+const allEnemies = [];
+
+const WIDTH_OF_GAME_SPRITES = 100;
+const COLUMNS_ON_GAME_FIELD = 5;
+const HEIGH_OF_GAME_FIELD_CELL = 83;
+const PLAYER_BUG_CENTERS_DISTANCE_OF_TOUCHING = 80;
+const Y_OF_GAME_FIELD_ROWS = [63, 146, 229, 312, 395];
+const Y_OF_LOWEST_GAME_FIELD_ROW = Y_OF_GAME_FIELD_ROWS[4];
+const MIN_SPEED_MULTIPLIER_FOR_BUG = 200;
+const MAX_SPEED_MULTIPLIER_FOR_BUG = 350;
+
+function resetPlayer(){
+    player.x = player.xInitial;
+    player.y = player.yInitial;
+}
+
+function resetEnemies(){
+    allEnemies.forEach((el)=>{        
+        el.x = el.xInitial;
+        el.speedKoef = Math.floor(Math.random() * MAX_SPEED_MULTIPLIER_FOR_BUG - MIN_SPEED_MULTIPLIER_FOR_BUG) + MIN_SPEED_MULTIPLIER_FOR_BUG;  
+    })
+}
+
+function isOnWater(){                   
+    if (this.y === this.yAxisMin){
+        resetEnemies();
+        resetPlayer();
+    }
+}
+
+
+var Enemy = function(y) {                    
+    this.sprite = 'images/enemy-bug.png';
+    this.speedKoef = Math.floor(Math.random() * MAX_SPEED_MULTIPLIER_FOR_BUG - MIN_SPEED_MULTIPLIER_FOR_BUG) + MIN_SPEED_MULTIPLIER_FOR_BUG;  
+    this.xInitial = -WIDTH_OF_GAME_SPRITES;
+    this.xAxisMin = this.xInitial;
+    this.x = this.xInitial;
+    this.xAxisMax = WIDTH_OF_GAME_SPRITES * COLUMNS_ON_GAME_FIELD;
+    this.y = y;
+};
+
+Enemy.prototype.doesHitPlayer = function (){
+    if (Math.abs(this.x - player.x) <= PLAYER_BUG_CENTERS_DISTANCE_OF_TOUCHING && this.y === player.y){   
```

Comment:
>     * [ ] classes do not refer to any global variables, like global variable `player`, which is an instance of `Player` class
      (referring to global constants and globals provided by the gaming platform like `Resources` is OK);
      Hint: pass `Player` instance as an argument to every enemy